mesonbuild/backend/backends.py

```diff
@@ -820,9 +905,9 @@ def generate_basic_compiler_args(self, target: build.BuildTarget, compiler: 'Com
         # command-line or default_options inside project().
         commands += compiler.get_option_compile_args(copt_proxy)
         # Add buildtype args: optimization level, debugging, etc.
-        commands += compiler.get_buildtype_args(self.get_option_for_target(OptionKey('buildtype'), target))
-        commands += compiler.get_optimization_args(self.get_option_for_target(OptionKey('optimization'), target))
-        commands += compiler.get_debug_args(self.get_option_for_target(OptionKey('debug'), target))
+        commands += compiler.get_buildtype_args(T.cast(str, self.get_option_for_target(OptionKey('buildtype'), target)))
+        commands += compiler.get_optimization_args(T.cast(str, self.get_option_for_target(OptionKey('optimization'), target)))
+        commands += compiler.get_debug_args(T.cast(bool, self.get_option_for_target(OptionKey('debug'), target)))
```

Comment:
I would prefer asserts here. There is no reason that the types should ever accidentally change, but they do for some reason we will be glad that the asserts tell us early where the problem is.